src/engine/data_gen.c

```diff
@@ -211,6 +289,22 @@ data_network_inputs_create(Memory* memory, DataSample* sample, Network* network,
             input->type = INPUT_SPIKES;
             input->spikes.spikes = spikes;
             input->spikes.n_spikes = layer->n_neurons;
+        } else if (sample->type == DATA_SAMPLE_ROC) {
+            // TODO: aici poate fi bug ca eu am un singur layer per sample, maybe add a check or something
+            DataSampleRoc* roc_sample = &(sample->roc_spikes);
+
+            input->type = INPUT_SPIKES;
+            // TODO: why spikes is b32 not b8?
```

Comment:
Yep, why do I use b32? I mean its not a big deal, but still a lot of waste for a a spike inputs, check if you can just use b8 and be done with it

Reply:
So, the b32 is not used in that many places so its not a huge difference and the bin files dumped don't go down in size